### ISSUE TITLE
Screen doesnt locked

### ISSUE THREAD

**Issue by Tatikoma** [celluloid-player/celluloid#532]
Overview Description:
After movie ends the screen doesnt locked even after few hours.

Steps to Reproduce:
1. Play movie.
2. Wait for movie ends
3. Continue waiting..

Actual Results: Screen locked

Expected Results: Screen not locked

Version: Celluloid 0.17, Linux Mint 19.3, kernel 5.3.0-40-generic #32~18.04.1-Ubuntu.

Additional Information:
Due to the fact that the screen does not lock, it does not turn off, which is why the projector connected to the computer does not turn off. I fall asleep under the movie, and the projector remains on all night and consumes resources.

**Comment by gnome-mpv** [comment]
I can't reproduce this.

I tested with Linux Mint 19.3 (Cinnamon) in a VM. The screen locks as expected a while after the video ends.

**Comment by Tatikoma** [comment]
It's very strange, because my installation about 1 month old. The problem appeared immediately after clean LM 19.3 installation. I have 2 mirrored displays.

I playing video fullscreen and after video ends i see not locked full black screen.

How i can collect and provide additional logs?

**Comment by gnome-mpv** [comment]
> How i can collect and provide additional logs?

1. Run `G_MESSAGES_DEBUG=all celluloid --mpv-options='--msg-level=all=trace' | tee /tmp/celluloid.log`
2. Play a video and wait for it to end.
3. Wait further for however long it usually takes for the lock screen to trigger.
4. Upload `/tmp/celluloid.log` here.

**Comment by Tatikoma** [comment]
[celluloid.log](https://github.com/celluloid-player/celluloid/files/4335019/celluloid.log)
Here is my log. I was waited for 1 hour without luck.

Screen lock timer is default, 15 minutes.

Without celluloid screen lock works fine.

**Comment by MedicMomcilo** [comment]
On Fedora 31, Gnome on Wayland, this doesn't occur even if we specifically note `stop-screensaver="no"` in config. I've even configured shortcut key for cycling between two options and had OSD confirmation that screensaver shouldn't be disabled.

Is there anything else that can be provided to aid in resolving this?

**Comment by gnome-mpv** [comment]
> Is there anything else that can be provided to aid in resolving this?

Are you using v0.18? If so, can you create a log file using the instructions above and upload it here? v0.18's log output is a bit more detailed so there might be something interesting in there.

**Comment by MedicMomcilo** [comment]
Indeed using 0.18:
```
[momcilo@localhost ~]$ rpm -q celluloid
celluloid-0.18-1.fc31.x86_64
[momcilo@localhost ~]$ 
```
For this specific test, screen blanking is set to 1minute and confirmed working without celluloid (and also with celluloid idly running without any media loaded).
Attaching celluloid.log (as well as celluloid-stderr.log which contains ffmpeg and other debugging info).

[celluloid.log](https://github.com/celluloid-player/celluloid/files/4438389/celluloid.log)
[celluloid-stderr.log](https://github.com/celluloid-player/celluloid/files/4438392/celluloid-stderr.log)

**Comment by gnome-mpv** [comment]
@MedicMomcilo From your log, it looks like the video never stopped playing until the end. Did you wait for the video to finish playing or pause it before the 1-minute countdown?

**Comment by MedicMomcilo** [comment]
@gnome-mpv no, video kept playing.
I'm sorry if I wasn't clear that my issue slightly differs from original report.
I just tested original scenario, and screen correctly blanks after the playback stops.

**Comment by MedicMomcilo** [comment]
Would you prefer me opening separate issue for my report?

**Comment by MedicMomcilo** [comment]
> > On Fedora 31, Gnome on Wayland, this doesn't occur even if we specifically note stop-screensaver="no" in config. I've even configured shortcut key for cycling between two options and had OSD confirmation that screensaver shouldn't be disabled.
> 
> Currently that option has no effect with Celluloid. Screen locking is always inhibited as long as there's a file playing.
> 
<snip>

Oooh, I completely missed this line.

Can this be implemented?
Possibly tieing to mpv setting "stop-screensaver".

**Comment by gnome-mpv** [comment]
> Would you prefer me opening separate issue for my report?

Yes.

> Can this be implemented?
> Possibly tieing to mpv setting "stop-screensaver".

Yes. It should be pretty easy to implement.

**Comment by Tatikoma** [comment]
I was updated libmpv (and some other packages, but i think libmpv) and now screen lock works fine.

add-apt-repository ppa:mc3man/bionic-media
apt-get update && apt-get upgrade

Before upgrade libmpv1:amd64 0.27.2-1ubuntu1
After upgrade libmpv1:amd64 2:0.30.0~bionic

I also tried to install latest celluloid version with flatpak, but it not worked before i upgraded libmpv. After upgrade both versions works fine.
